src/main/java/com/iota/iri/service/milestone/impl/MilestoneSolidifierImpl.java

```diff
@@ -205,9 +206,7 @@ private void processSolidifyQueue() throws Exception {
                     }
                     break;
                 case INCOMPLETE:
-                    if(milestoneIndex > getLatestMilestoneIndex()) {
-                        registerNewMilestone(getLatestMilestoneIndex(), milestoneIndex, milestoneHash);
-                    }
+                    registerNewMilestone(getLatestMilestoneIndex(), milestoneIndex, milestoneHash);
```

Comment:
why are you also registering milestones that may be fake?

Reply:
The check has been moved to `registerNewMilestone`